pkg/jsonnet/jpath/dirs.go

```diff
@@ -0,0 +1,110 @@
+package jpath
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+)
+
+// FindRoot returns the absolute path of the project root, being the directory
+// that directly holds `tkrc.yaml` if it exists, otherwise the directory that
+// directly holds `jsonnetfile.json`
+func FindRoot(path string) (dir string, err error) {
+	start, err := FsDir(path)
+	if err != nil {
+		return "", err
+	}
+
+	// root path based on os
+	stop := "/"
+	if runtime.GOOS == "windows" {
+		stop = filepath.VolumeName(start) + "\\"
+	}
+
+	// try tkrc.yaml first
+	root, err := FindParentFile("tkrc.yaml", start, stop)
+	if err == nil {
+		return root, nil
+	}
+
+	// otherwise use jsonnetfile.json
+	root, err = FindParentFile("jsonnetfile.json", start, stop)
```

Comment:
nit: Can we have list of `parentFiles []string` and iterate over it? It could make the `jsonnet/jpath` API a bit more flexible.

Reply:
Where would that go? I would not want every callee of `jpath.Resolve` or `jpath.FindRoot` have to specify that

Reply:
Merging, we can address that later if we wish to